pkg/cloudprovider/providers/aws/aws.go

```diff
@@ -1090,7 +1112,7 @@ func (c *Cloud) getAllZones() (sets.String, error) {
 // GetZone implements Zones.GetZone
 func (c *Cloud) GetZone() (cloudprovider.Zone, error) {
 	return cloudprovider.Zone{
-		FailureDomain: c.selfAWSInstance.availabilityZone,
+		FailureDomain: c.availabilityZone,
```

Comment:
I don't think this is what we want.  It appears to be used in two places:

```
pkg/controller/service/servicecontroller.go:    zone, err := zones.GetZone()
pkg/kubelet/kubelet_node_status.go:                     zone, err := zones.GetZone()
```

For kubelet node status, we want to continue to label with the node's zone, not the master's zone.

For servicecontroller, zone is actually unused since be9ce3089779c4aa32b0744e8443f169a2d6bf19 I believe.  I have proposed it for removal in #40060

Hence I think we want to keep it returning selfAWSInstance.